cogs/birthday.py

```diff
@@ -0,0 +1,97 @@
+import datetime
+import re
+from typing import Pattern
+
+from discord.ext import commands
+
+
+class Birthdays(commands.Cog):
+    DATEPATTERN: Pattern[str] = re.compile(r'(((0[1-9])|(1[0-9])|(2[0-9])|(3[0-1]))\.'  # 01.-31.
+                                           r"((01)|(03)|(05)|(07)|(08)|(10)|(12))\.)"  # all months with 31 days
+                                           r"|"
+                                           r"(((0[1-9])|(1[0-9])|(2[0-9])|(30))\."  # 01.-30.
+                                           r"((04)|(06)|(09)|(11))\.)"  # all months with 30 days
+                                           r"|"
+                                           r"(((0[1-9])|(1[0-9])|(2[0-9]))\."  # 01.-29.
+                                           r"02\.)")  # february
+    BIRTHDAY_CHANNEL_ID = 666653781366145028
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.bot.register_job(60 * 60 * 24, self.congratulate)
+
+    @commands.group(aliases=['birth', 'birthday', 'birthdate', 'geburtstag'])
+    async def birthdays(self, ctx):
+        if ctx.invoked_subcommand is None:
+            await ctx.send("Ungültiger command!")
+
+    @birthdays.command()
+    @commands.has_permissions(manage_messages=True)
+    async def list(self, ctx, querry=""):
+        """lists all birthdays (birthdays or userid are possible, jet optional querries)"""
+        with self.bot.db.get(self.BIRTHDAY_CHANNEL_ID) as db:
+            if len(querry) is 0:  # No Querry
+                results = db.execute("SELECT userId, date, month FROM birthdays").fetchall()
+                text = ""
+                for result in results:
+                    text += "User: {}\t->\t{}.{}.\n".format(result[0], result[1], result[2])
+                await ctx.send(text)
+            elif self.DATEPATTERN.match(querry) is not None:  # Birthday as Querry
+                day, month = querry.strip(".").split(".")
+                results = db.execute(
+                    "SELECT userId, date, month FROM birthdays WHERE date = {} AND month = {}".format(day,
```

Comment:
Please use prepared statements, which are less prone to injections.

Reply:
sorry I totally forgot about those